crates/bevy_math/src/geometry.rs

```diff
@@ -25,17 +55,55 @@ impl<T: Default + Reflect + PartialEq> Default for Size<T> {
     }
 }
 
-/// A rect, as defined by its "side" locations
+/// A rectangle defined by its side locations.
+///
+/// It is commonly used to define positions, margins, paddings and borders.
+///
+/// # Example
+///
+/// ```rust
+/// # use bevy_math::Rect;
+/// #
+/// let rect = Rect::<f32> {
+///     left: 1.0,
+///     right: 2.0,
+///     top: 3.0,
+///     bottom: 4.0,
+/// };
+///
+/// assert_eq!(rect.left, 1.0);
+/// assert_eq!(rect.right, 2.0);
+/// assert_eq!(rect.top, 3.0);
+/// assert_eq!(rect.bottom, 4.0);
+/// ```
 #[derive(Copy, Clone, PartialEq, Debug, Reflect)]
 #[reflect(PartialEq)]
 pub struct Rect<T: Reflect + PartialEq> {
+    /// The left side of the rectangle.
     pub left: T,
+    /// The right side of the rectangle.
     pub right: T,
+    /// The top side of the rectangle.
     pub top: T,
+    /// The bottom side of the rectangle.
     pub bottom: T,
 }
 
 impl<T: Reflect + PartialEq> Rect<T> {
+    /// Creates a new [`Rect`] from a single value.
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// # use bevy_math::Rect;
+    /// #
+    /// let rect = Rect::<f32>::all(1.0);
+    ///
+    /// assert_eq!(rect.left, 1.0);
+    /// assert_eq!(rect.right, 1.0);
+    /// assert_eq!(rect.top, 1.0);
+    /// assert_eq!(rect.bottom, 1.0);
+    /// ```
     pub fn all(value: T) -> Self
```

Comment:
```suggestion
    pub fn all(distance_from_center: T) -> Self
```

When doing docs PRs, I often sneak in small variable renames like this that clarify intent.

Reply:
Good Point! Done.